exonum-java-binding-core/src/main/java/com/exonum/binding/blockchain/Blockchain.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.exonum.binding.blockchain;
+
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.storage.database.View;
+import com.exonum.binding.storage.indices.ListIndex;
+import com.exonum.binding.storage.indices.ProofListIndexProxy;
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Provides read-only access to the blockchain state that is maintained by Exonum core:
+ * blocks, transaction messages, execution results. Please refer to the
```

Comment:
I'd reword that "This class is equivalent to (or provides access to) the blockchain::Schema in the Rust API."
 because it's not OK to refer Java users to that documentation unconditionally — it won't make any sense to them.

Reply:
Why don't we leave possibility for a user to go to the documentation of Rust `blockchain::Schema` if we refer to it anyway? Right now this javadoc doesn't seem clear to me. I think it's better to have at least some information than none.
Or maybe we should refer to `CoreSchemaProxy` here?

Reply:
I might not have been clear: it's good to have a link to the core schema, but the wording of the second sentence is better be improved.

>  Right now this javadoc doesn't seem clear to me.

Please request improving it (ideally, with suggestions) — it must be clear :upside_down_face:

Reply:
I think now it's okay, but I'd still add a link to it:
```
 * Provides read-only access to the subset of <a href="https://docs.rs/exonum/latest/exonum/blockchain/struct.Schema.html">blockchain::Schema</a> features in the Core API:
 * blocks, transaction messages, execution results.
```
Something like that?